lib/serializer.ts

```diff
@@ -165,35 +165,48 @@ export class Serializer {
       objectName = mapper.serializedName!;
     }
 
-    if (mapperType.match(/^Number$/ig) !== null) {
+    if (mapperType.match(/^Composite$/ig) !== null) {
+      payload = deserializeCompositeType(this, mapper as CompositeMapper, responseBody, objectName);
+    } else if (mapperType.match(/^Number$/ig) !== null) {
       payload = parseFloat(responseBody);
       if (isNaN(payload)) {
         payload = responseBody;
       }
-    } else if (mapperType.match(/^Boolean$/ig) !== null) {
-      if (responseBody === "true") {
-        payload = true;
-      } else if (responseBody === "false") {
-        payload = false;
-      } else {
+    } else {
+      if (this.isXML) {
+        /**
+         * If the mapper specifies this as a non-composite type value but the responseBody contains
+         * both header ("$") and body ("_") properties, then just reduce the responseBody value to
+         * the body ("_") property.
+         */
+        if (responseBody["$"] != undefined && responseBody["_"] != undefined) {
+          responseBody = responseBody["_"];
+        }
+      }
+
+      if (mapperType.match(/^Boolean$/ig) !== null) {
+        if (responseBody === "true") {
```

Comment:
Can that be simplified to:?
```suggestion
if (responseBody) {
  payload = responseBody;
}
```